### PR TITLE
Apply: user sign in/up flow

### PR DESCRIPTION
**Pull request by JoshBroomberg**
Adds omniauth to the user with original login view. 

Depends on #6

### CODE REVIEW

File: app/controllers/devise/custom_omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Devise::CustomOmniauthCallbacksController < ::Devise::OmniauthCallbacksController
+  def facebook
+    @user = User::AsOmniauthable.from_omniauth(request.env["omniauth.auth"], current_user)
```
Comment (JoshBroomberg) [inline app/controllers/devise/custom_omniauth_callbacks_controller.rb]: I cleaned up this file upstream

File: app/models/submission.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Submission < ActiveType::Object
```
Comment (JoshBroomberg) [inline app/models/submission.rb]: This file changes drastically upstream

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,27 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the rake db:seed (or created alongside the db with db:setup).
-#
-# Examples:
-#
-#   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
-#   Mayor.create(name: 'Emanuel', city: cities.first)
+# Destroy all models.
```
Comment (JoshBroomberg) [inline db/seeds.rb]: This file changes drastically upstream

File: app/models/user/as_omniauthable.rb [hunk 4]
```diff
@@ -0,0 +1,65 @@
+class User::AsOmniauthable < ActiveType::Record[User]
+  def self.from_omniauth(auth_data, current_user)
+    user = where(provider: auth_data.provider, uid: auth_data.uid)
+        .first_or_initialize
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: 2 spaces, not 4

File: app/models/user/as_omniauthable.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      
+    user.email = auth_data.info.email if user.email.blank?
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: Are we sure this is ok if `auth_data.info.email` is also blank?

Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: Also surely we should `||=` here?

File: app/models/user/as_omniauthable.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    user.password ||= Devise.friendly_token[0,20]
+
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: Why this newline?

File: app/models/user/as_omniauthable.rb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  def self.from_omniauth(auth_data, current_user)
+    user = where(provider: auth_data.provider, uid: auth_data.uid)
+        .first_or_initialize
+      
+    user.email = auth_data.info.email if user.email.blank?
+    user.password ||= Devise.friendly_token[0,20]
+    
+    user.first_name ||= auth_data.info.first_name
+    user.last_name ||= auth_data.info.last_name
+
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: I'd get rid of all these new lines.

File: app/models/user/as_omniauthable.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    user.fb_link = auth_data.info.urls.Facebook
+    # TODO (josh): add picture to user.
+    #user.picture = process_uri((auth.info.image)) if auth.info.image    
+    
+    user.gender ||= auth_data.extra.raw_info.gender
+    
+    if auth_data.extra.raw_info.hometown
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: You don't need this if statement with an `||=`

File: app/models/user/as_omniauthable.rb [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    if auth_data.extra.raw_info.location
+      user.current_city ||= auth_data.extra.raw_info.location["name"]
+    end
+
+    user.birthday ||= process_date(auth_data.extra.raw_info.birthday)
+    edu_params = process_education(auth_data.extra.raw_info.education)
+    # TODO (josh): add university info to user.
+    #user.university_id ||= edu_params[0]
+    #user.graduation_year ||= edu_params[1]
+    user.save
```
Comment (khanotations) [inline app/models/user/as_omniauthable.rb]: Should this be a `save!`

**Pull request merged**